src/Parser/Warrior/Fury/Modules/Talents/FuriousSlashTimesByStacks.js

```diff
@@ -0,0 +1,102 @@
+import Analyzer from 'Parser/Core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import HIT_TYPES from 'Parser/Core/HIT_TYPES';
+
+/* 
+  furiousSlashTimesByStacks() returns an array with the durations of each FS buff stack
+*/
+
+const MAX_FURIOUS_SLASH_STACKS = 3;
+
+class FuriousSlashTimesByStacks extends Analyzer {
+	
+	furiousSlashStacks = [];
+	lastFuriousSlashStack = 0;
+	lastFuriousSlashUpdate = this.owner.fight.start_time;
+	
+	constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.FURIOUS_SLASH_TALENT.id);
+	this.furiousSlashStacks = Array.from({length: MAX_FURIOUS_SLASH_STACKS + 1}, x => []);
+  }
+	
+	handleStacks(event, stack = null) {
+    if (event.type === 'removebuff' || isNaN(event.stack)) { //NaN check if player is dead during on_finish
+      event.stack = 0;
+    }
+	/*
+    if (event.type === 'applybuff') {
+      event.stack = 1;
+    }
+	*/
+	
+	if (event.type === 'damage') {
+		if(!this.lastFuriousSlashStack)
+		{
+			event.stack = 1;
+		}
+		else{
+			if(this.lastFuriousSlashStack < MAX_FURIOUS_SLASH_STACKS)
+			{
+				event.stack = this.lastFuriousSlashStack + 1;
+			}
+			else{
+				event.stack = MAX_FURIOUS_SLASH_STACKS;
+			}
+		}
+	}
+	
+    if (stack) {
+      event.stack = stack;
+    }
+
+    this.furiousSlashStacks[this.lastFuriousSlashStack].push(event.timestamp - this.lastFuriousSlashUpdate);
+    this.lastFuriousSlashUpdate = event.timestamp;
+    this.lastFuriousSlashStack = event.stack;

```

Comment:
Does this only occur once per fight? If I am following the code correctly this looks like it only works once.